lib/pengine/unpack.c

```diff
@@ -2024,6 +2025,8 @@ process_rsc_state(pe_resource_t * rsc, pe_node_t * node,
     }
 
     if (node->details->unclean) {
+        /* In order to calculate priority_fencing_delay correctly, save the failure information and pass it to native_add_running(). */
+        save_on_fail = on_fail;
```

Comment:
I think we want to do this even if the node is clean. If the node is clean but a resource is failed on it, I think it makes sense not to count the resource toward the node priority.

Reply:
Okay!

I think it's correct.
Move on_fail save before unclean decision.